File: 2026POTD/JANUARY/10-01-2026/712.rb
def minimum_delete_sum(s1, s2)
    n, m = s1.length, s2.length
    dp = Array.new(n + 1) { Array.new(m + 1, 0) }

    (n - 1).downto(0) do |i|
        dp[i][m] = dp[i + 1][m] + s1[i].ord
    end

    (m - 1).downto(0) do |j|
        dp[n][j] = dp[n][j + 1] + s2[j].ord
    end

    (n - 1).downto(0) do |i|
        (m - 1).downto(0) do |j|
            if s1[i] == s2[j]
                dp[i][j] = dp[i + 1][j + 1]
            else
                dp[i][j] = [
                    s1[i].ord + dp[i + 1][j],
                    s2[j].ord + dp[i][j + 1]
                ].min
            end
        end
    end

    dp[0][0]
end

File: 2026POTD/JANUARY/10-01-2026/712.rs
impl Solution {
    pub fn minimum_delete_sum(s1: String, s2: String) -> i32 {
        let a = s1.as_bytes();
        let b = s2.as_bytes();
        let n = a.len();
        let m = b.len();
        let mut dp = vec![vec![0; m + 1]; n + 1];

        for i in (0..n).rev() {
            dp[i][m] = dp[i + 1][m] + a[i] as i32;
        }

        for j in (0..m).rev() {
            dp[n][j] = dp[n][j + 1] + b[j] as i32;
        }

        for i in (0..n).rev() {
            for j in (0..m).rev() {
                if a[i] == b[j] {
                    dp[i][j] = dp[i + 1][j + 1];
                } else {
                    dp[i][j] = std::cmp::min(
                        a[i] as i32 + dp[i + 1][j],
                        b[j] as i32 + dp[i][j + 1],
                    );
                }
            }
        }

        dp[0][0]
    }
}